### PR TITLE
fix: should reset frameBytesLeft after writing

### PR DESCRIPTION
**Pull request by Sniper91**
Signed-off-by: sniper91 <kevinzhao91@outlook.com>

### CODE REVIEW

**Review by GiedriusS — APPROVED** [review]
:+1: good catch, should reduce the number of write() significantly

File: storage/remote/codec_test.go [hunk 1]
```diff
@@ -367,3 +370,119 @@ func TestNilHistogramProto(t *testing.T) {
...
+}
+
+type mockWriter struct {
+	actual []*prompb.ChunkedSeries
+}
+
+func (m *mockWriter) Write(p []byte) (n int, err error) {
+	cr := &prompb.ChunkedReadResponse{}
+	if err := proto.Unmarshal(p, cr); err != nil {
+		return 0, nil
```
Comment (GiedriusS) [inline storage/remote/codec_test.go:425]: I suggest returning the error here:
```suggestion
		return 0, fmt.Errorf("unmarshaling: %w", err)
```

Comment (Sniper91) [reply storage/remote/codec_test.go]: It was fixed.Thanks

**Review by GiedriusS — APPROVED** [review]
I'm not a maintainer so I can't merge your PR but I've pinged people on Slack :crossed_fingers:

**Review by bwplotka — APPROVED** [review]
Nice! Thanks, nice catch 💪🏽

**Pull request merged**